pymc3/distributions/dist_math.py

```diff
@@ -139,19 +139,85 @@ def log_normal(x, mean, **kwargs):
         std = rho2sd(rho)
     else:
         std = tau**(-1)
-    std += f(eps)
-    return f(c) - tt.log(tt.abs_(std)) - (x - mean) ** 2 / (2. * std ** 2)
-
+    std += floatX(eps)
+    return floatX(c) - tt.log(tt.abs_(std)) - (x - mean) ** 2 / (2. * std ** 2)
+
+
+def CholeskyCheck(mode='cov', return_ldet=True, replacement=None):
+    """Checks if the given matrix/cholesky is positive definite. Returns a dummy
+       Cholesky replacement if it is not, along with a boolean to assert whether
+       replacement was needed and, optionally, the log of the determinant of
+       either the real Cholesky or its replacement."""
+    is_cholesky = (mode == 'chol')
+    w_ldet = return_ldet
+    # add inplace=True when/if impletemented by Theano
+    cholesky = slinalg.Cholesky(lower=True, on_error="nan")
+
+    # Check if a given Cholesky is positive definite
+    def check_chol(cov):
```

Comment:
I'm not sure we actually need this check. The cholesky decomposition is only defined for positive definite matrices in the first place. So we could argue that it's the users fault if they pass in something that is not actually a cholesky matrix. (The return vals of the cholesky op should always have a positive diagonal.)

Reply:
Right. With GP no longer passing sampled chol's, only LKJ would, and LKJ wouldn't pass something with neg diagonal. I think that stands so I'll remove.

Reply:
@ferrine I think FullRankGroup is your domain, I'm afraid it's out of my scope of expertise. Is there a reason the L property might return non-positive or null diagonals? Could a check for that be implemented there? Would it be ok to add 1e-10's to the diagonal?